### ISSUE TITLE
Healthcheck for Kubernetes

### ISSUE THREAD

**Issue by aleksasiriski** [SenseUnit/dumbproxy#70]
Hello there, I'm using this project within Kubernetes and wanted to know if there is any path I can use for healthcheck and liveness probes in Kubernetes?

**Comment by Snawoot** [comment]
Hello!

It has no endpoint for healthcheck at this moment. But if you're using alpine-based image, you can probably define some healthcheck command to check if proxy responds. One way to check it is to use OPTIONS request like this: `curl -v --request-target "*" -X OPTIONS http://127.0.0.1:8080`

**Comment by aleksasiriski** [comment]
Interesting idea, but I would prefer to stick with the scratch image as it has less attack surface. Would you consider adding a simple HTTP 200 OK endpoint on `/healthz` and another when the initialization is done on `/readyz` for K8s use cases? If you want I can make a PR for it.

**Comment by Snawoot** [comment]
It is a good solution to me, I'll be glad to accept PR. But let's discuss few questions:

* Initialization is done as soon as HTTP server started, so probably /healthz and /readyz would mean the same. Maybe we need just one endpoint for all probes?
* It is very undesirable to have any static signatures which can help to identify dumbproxy with active probes. Maybe we can make healthcheck URI configurable and enabled only when specified explicitly. Maybe we can require some secret header or something like that.
* How will we distinguish between proxy requests with full URI in the request line and healthcheck request with full URI in the request line (uncommon, but still allowed by standard)? Maybe some marker headers is the way to distinguish them reliably?

**Comment by aleksasiriski** [comment]
Hey, sorry for the delay, here are my thoughts:

1) Could very well be the same, I will look at the code and think about this more
2) I would be okay with having this be an opt-in flag / env var, but in Kubernetes you wouldn't expose those paths, only the proxy one to the outside
3) Not completely sure what you mean, I will take a look at the code and we can discuss more

**Comment by aleksasiriski** [comment]
Btw, I ended up making my own thing (will edit this comment with the link since it'll be open source soon, if you want to see it), mainly because when I looked at your code I couldn't really understand what was happening and where, after a while I came to the conclusion that I dislike the use of non-stdlib functions and additional features that, for my use case, aren't necessary. As for the points:

1. `/healthz` should always return 200 OK (it means that the webserver has started), whereas `/readyz` should return 200 OK if it is able to resolve some test URL (like `https://google.com` or `https://cloudflare.com/cdn-cgi/trace`)
2. I only use the proxy via K8s service, and it works fine.
3. I ended up having a separate http server running on a different port for the liveness and readiness probes, and now admins just have to leave it locked within the private network and it won't be reachable from the outside.
